### ISSUE TITLE
High CPU usage on a Rails 7.2 app using json_tagged_logger 0.8.0

### ISSUE THREAD

**Issue by santry** [santry/json_tagged_logger#21]

**Comment by santry** [comment]
@pboling are you noticing this on any of your apps that are using 0.8.0 (if you have any yet)?

**Comment by pboling** [comment]
We haven't released any with the new version yet.  Our change process is pretty slow (health care).

It doesn't seem like there are any changes between 0.7.0 and 0.8.0 that could significantly affect CPU usage though.

**Comment by santry** [comment]
I agree, there's nothing in the change that I would expect to affect CPU. Even so, on one of my high-volume apps the CPU jumped from baseline to about 2x baseline. Reverting to 0.7.0 fixed it.

<img width="1371" alt="Screenshot 2024-10-26 at 16 36 20" src="https://github.com/user-attachments/assets/db51f628-6399-4240-a504-49c9c71c77f9">

I'll do some experimentation to see if I can find the cause but if not, I may release a new version that backs out the change. You'll still be able to pin your apps to 0.8.0 to get the older Rails support. Unless, of course, you start to see the problem too, in which case I'll need to debug further.

**Comment by pboling** [comment]
Maybe another gem was also updated?  JSON gem has seen a bunch of updates recently?  My guess would be that it has nothing to do with this.

**Comment by pboling** [comment]
I'm guessing this may have more to do with the `json` gem being updated relentlessly (and again today) with new bugs and new bug fixes.

**Comment by pboling** [comment]
@santry did you ever find a culprit here?

We will be launching the latest version into a high traffic env within a week!
